### PR TITLE
WL-2421: Get the contenthandler for the resource type and call copy on it

### PR DESCRIPTION
**Pull request by nicholaswilson100**
I've created an interface for all ContentChangeHandlers to implement and the ability to register a handler with a resource type.

### CODE REVIEW

**Comment by nicholaswilson100** [comment]
This fix relies on WL-3903 being in.

**Comment by nicholaswilson100** [comment]
Linked to https://github.com/ox-it/wl-citations/pull/15

**Comment by buckett** [comment]
Can you JavaDoc the new classes and methods about what they are for and why you might want them

File: api/src/main/java/org/sakaiproject/content/api/ContentChangeHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.sakaiproject.content.api;
+
+public class ContentChangeHandlerImpl implements ContentChangeHandler {
```
Comment (buckett) [inline api/src/main/java/org/sakaiproject/content/api/ContentChangeHandlerImpl.java:3]: If you're doing this for interface evolution Java 8 default methods go some way to handling this and might be better?

**Comment by buckett** [comment]
Just discussed this in the office and going to change `BaseContentService.copyResource()` to have the change handler code in it. Then add a hack when duplicating sites to call the raw copy rather than doing it's add/update.

Need to keep in mind that in the future we might want to move the duplicate hack into a `ContentChangeHandler`

**Pull request closed without merging**